R/audit_deps.R

```diff
@@ -73,6 +76,22 @@ audit_deps = function(pkgs = NULL, verbose = TRUE) {
   return(results)
 }
 
+
+#' Check PyPi package dependencies
+#'
+#' Search the OSS Index for known PyPi software dependency vulnerabilities.
+#'
+#' @param pkg Name of package as listed on PyPi.
+#' @param version The version of the package. Optional.
+#' @param verbose Default \code{TRUE}
+#'
+#' @details By default `audit_pypi_deps()` will return audits for all known versions of a package on PyPi. To search for a single package version, supply the version as a character to the `version` argument.
+
+audit_pypi_deps <- function(pkg, version = "*", verbose = TRUE) {
```

Comment:
I'm not really keen on this implementation. It's doesn't extend that nicely. What about when you call `audit_deps()`

- if pkgs is `NULL`, then look at `installed.packages()` - current behaviour
- if you pass a data.frame, we can include an additional column called purl. If purl was `pypi`, we just add the `pkg:pypi/`

How does that sound?

Reply:
I agree with extending to other pkg purls. 

I think my hesitance is in passing a whole data frame and inferring arguments from that data frame (assumedly based on column names). I'd rather an implementation that accepts vectors for each argument—e.g. `pkg`, `version`, `scheme`. This way you can run audit deps inside of call like `mutate(df, pkg_audit = audit_pkg_deps(pkg_col, version_col, scheme_col))`. This would also give you more flexibility in being able to use list objects etc.

Part of the rationale here is I think it makes sense to be able to audit packages _before_ they're installed. 

Alternatively this could be a single function which checks if `pkg` is a data.frame, if so use the `installed.packages()` approach. Then, use a a vector based approach.